uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/TOILRequest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequestDate as LeaveRequestDate;
+use CRM_HRLeaveAndAbsences_BAO_TOILRequest as TOILRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequest as LeaveRequestService;
+
+class CRM_HRLeaveAndAbsences_Service_TOILRequest extends LeaveRequestService {
+
+  /**
+   * This method creates/updates the TOILRequest along with the
+   * associated LeaveRequest and it's LeaveBalanceChanges
+   *
+   * @param array $params
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_TOILRequest|NULL
+   */
+  protected function createRequestWithBalanceChanges($params) {
+    $toilRequest = TOILRequest::create($params, false);
+    $expiryDate = !empty($params['expiry_date']) ? new DateTime($params['expiry_date']) : null;
+
+    $this->leaveBalanceChangeService->createForTOILRequest(
+      $toilRequest,
+      $params['type_id'],
+      $params['toil_to_accrue'],
+      $expiryDate
+    );
+
+    return $toilRequest;
+  }
+
+  /**
+   * Returns the LeaveRequest object associated with the TOILRequest
+   * in its current state (i.e before it gets updated)
+   *
+   * @param int $toilRequestID
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_LeaveRequest
+   */
+  protected function getOldLeaveRequest($toilRequestID) {
+    $toilRequest = TOILRequest::findById($toilRequestID);
+    if (!$this->oldLeaveRequest) {
+      $this->oldLeaveRequest = LeaveRequest::findById($toilRequest->leave_request_id);
+    }
+    return $this->oldLeaveRequest;
+  }
+
+  /**
+   * Deletes the TOILRequest with the given $toilRequestID, including
+   * its associated LeaveRequest and all of its LeaveRequestDates and
+   * TOIL LeaveBalanceChanges
+   *
+   * @param int $toilRequestID
+   */
+  public function delete($toilRequestID) {
+    $toilRequest = TOILRequest::findById($toilRequestID);
+    $leaveRequest = LeaveRequest::findById($toilRequest->leave_request_id);
+
+    $transaction = new CRM_Core_Transaction();
+    try {
+      $this->leaveBalanceChangeService->deleteForTOILRequest($toilRequest);
+      LeaveRequestDate::deleteDatesForLeaveRequest($toilRequest->leave_request_id);
+      $leaveRequest->delete();
```

Comment:
Thought of that too but there was an extra logic in the parent method to delete LeaveBalance changes for the Leave Request. Since LeaveBalance changes for TOIL is created differently, I thought I should do it this way. although I know calling the parent::delete function won't break the function. What do you think?